### ISSUE TITLE
FHIR Server ARM Template deployments fails

### ISSUE THREAD

**Issue by arieschwartzman** [microsoft/fhir-server#877]
**Describe the bug**
Running the following fails:
New-AzResourceGroupDeployment -TemplateUri https://raw.githubusercontent.com/Microsoft/fhir-server/master/samples/templates/default-azuredeploy-sql.json -ResourceGroupName $rg.ResourceGroupName -serviceName $fhirServiceName

I'm getting this error:
New-AzResourceGroupDeployment : The template deployment 'default-azuredeploy' is not valid according to the validation
procedure. The tracking id is '1cbe52c4-3b5c-446c-9926-c4b95bcacf2f'. See inner errors for details.
At line:1 char:1
+ New-AzResourceGroupDeployment -TemplateUri https://raw.githubusercont ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [New-AzResourceGroupDeployment], CloudException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.NewAzureResourceGroupDeploymentCmd     let

Getting the extended error using the correlation Id using Get-AzLog, I'm getting this error
statusCode     : BadRequest
                       serviceRequestId:
                       statusMessage  : {"error":{"code":"InvalidTemplateDeployment","message":"The template deployment
                       'default-azuredeploy' is not valid according to the validation procedure. The tracking id is
                       '1cbe52c4-3b5c-446c-9926-c4b95bcacf2f'. See inner errors for
                       details.","details":[{"code":"ValidationForResourceFailed","message":"Validation failed for a resource.                           Check 'Error.Details[0]' for more information.","details":[{"code":"ServerFarmNotFound","message":"The                            specified app service plan was not found."}]}]}}

**Note:**
If I copy this template and deploy via the Azure portal, it works fine

**Comment by hansenms** [comment]
I have confirmed this bug, however, it is a bit puzzling since we have been running with this for a while, so it could be something that changed on the ARM side of things. We will have to investigate. 

@brendankowitz we made some changes to use the zipdeploy and there was a quick fix for the dependencies there to avoid that small problem we had, but as far as I can see, we did not touch the dependencies between appservice plan and app service.

**Comment by brendankowitz** [comment]
Don't think its anything we did, I get the same error using the version before any changes. I recall seeing an issue like this for ARM in the past

```
New-AzResourceGroupDeployment -TemplateUri https://raw.githubusercontent.com/microsoft/fhir-server/6d9ed8bcaf9f5ba18c683da8841c31ac9b11374d/samples/templates/default-azuredeploy.json
```

**Comment by hansenms** [comment]
I agree, something changed in ARM. Last time we had this issue, I think @jackliums was involved in our workaround. Maybe some learnings from that?

**Comment by brendankowitz** [comment]
For now, the workaround in this issue (https://github.com/microsoft/botframework-solutions/issues/2762) also seems to work here.

So changing this https://github.com/microsoft/fhir-server/blob/master/samples/templates/default-azuredeploy-sql.json#L204 to
```
"serverFarmId": "[concat('/subscriptions/', Subscription().SubscriptionId,'/resourcegroups/', resourceGroup().name, '/providers/Microsoft.Web/serverfarms/', variables('appServicePlanName'))]"
```

**Comment by uxbridgeroad** [comment]
The suggested workaround does not seem to work.
I made the suggested change to default-azuredeploy.json  and referenced that file locally :-

                "serverFarmId": "[concat('/subscriptions/', Subscription().SubscriptionId,'/resourcegroups/', resourceGroup().name, '/providers/Microsoft.Web/serverfarms/', variables('appServicePlanName'))]"


The following fails with exceptions:-

az group deployment create -g <resource_group> --template-file default-azuredeploy.json --parameters serviceName= fhirVersion=R4 applicationInsightsLocation=westeurope

Azure Error: InvalidTemplateDeployment
Message: The template deployment 'default-azuredeploy' is not valid according to the validation procedure. The tracking id is 'f3fa5996-4d0a-4693-81fa-f68764980877'. See inner errors for details.
Exception Details:
	Error Code: ValidationForResourceFailed
	Message: Validation failed for a resource. Check 'Error.Details[0]' for more information.

Any suggestions on a way forward?

**Comment by hansenms** [comment]
We will put up a workaround. We are working on it. In the meantime, if you deploy the template (as is) through the portal, it apparently works.

**Comment by uxbridgeroad** [comment]
Portal deployment worked, thanks :-)

I noticed that with the Msdeploy Package Url , we can specific a forked repo /branch. Previously in the KUDU version if the src ref. was left blank only the PaaS components were created (no app deployment). It was then possible to do an independent deployment. 

Is there an equivalent in this new version?

**Comment by brendankowitz** [comment]
@uxbridgeroad right now the options are to pass in a webdeploy file to use instead or it will use the latest build from master. After the deploy succeeds you can always push a deployment over the top.

**Comment by nikkh** [comment]
This morning I had to come back to one of my old demos.  It deploys using a DevOps pipeline, so I can delete it when I finish to save money.  I have deployed 161 times over the last couple of years and it has always worked.  This morning it failed.  Luckily I stumbled across this.  Changing as suggested above from subscription().id to Subscription().SubscriptionId in serverFarmId in my ARM template fixed it for me.
